spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/JavaClassMimeTypeUtils.java

```diff
@@ -74,7 +74,7 @@ public static MimeType mimeTypeFromObject(Object payload, String originalContent
 	 * Retrieve the class name from {@link MimeType}.
 	 *
 	 * @param mimeType {@link MimeType} to retrieve class name from
-	 * @return class name from mime type
+	 * @return class name from MimeType and null if the class name cannot be determined from the MimeType
```

Comment:
But you determine it from there only if there is `type` parameter.
Why don't mention that in JavaDocs?
This is public useful API, but a bit confusing without details.

Just IMHO though.

Feel free to leave as is.